api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -469,4 +466,82 @@ public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() throws Ex
 		new VisitValidator().validate(visit, errors);
 		assertEquals(true, errors.hasFieldErrors("voidReason"));
 	}
+
+	/**
+	 *  @see VisitValidator#validate(Object,Errors)
+	 */
+	@Test
+	@Verifies(value= "should fail validation if the visit start date is before the birth date of the patient", method = "validate(Object, Errors)")
+	public void validate_shouldFailValidationIfVisitStartDateIsBeforeBirthdate() throws Exception {
+		Calendar calendar = Calendar.getInstance();
+		calendar.set(1970, Calendar.FEBRUARY, 2);
+		Visit visit = makeVisit(42);
+		visit.setStartDatetime(calendar.getTime());
+
+		visit.setPatient(Context.getPatientService().getPatient(2));
+
+		Errors errors = new BindException(visit, "visit");
+		new VisitValidator().validate(visit, errors);
+
+		assertTrue(errors.hasFieldErrors("startDatetime"));
+	}
+
+	/**
+	 *  @see VisitValidator#validate(Object,Errors)
+	 */
+	@Test
+	@Verifies(value= "should fail validation if the visit start date is before the estimated birth date of the patient", method= "validate(Object, Errors)")
+	public void validate_shouldFailValidationIfVisitStartDateIsBeforeEstimatedBirthdate() throws Exception {
+		Calendar calendar = Calendar.getInstance();
+		calendar.set(1950, Calendar.FEBRUARY, 2);
+		Visit visit = makeVisit(42);
+		visit.setStartDatetime(calendar.getTime());
+
+		visit.setPatient(Context.getPatientService().getPatient(7));
+
+		Errors errors = new BindException(visit, "visit");
+		new VisitValidator().validate(visit, errors);
+
+		assertTrue(errors.hasFieldErrors("startDatetime"));
+	}
+
+	/**
+	 *  @see VisitValidator#validate(Object,Errors)
+	 */
+	@Test
+	@Verifies(value= "should fail validation if the visit start date is before the estimated birth date of the patient less than 2 YO", method= "validate(Object, Errors)")
+	public void validate_shouldFailValidationIfVisitStartDateIsBeforeEstimatedBirthdateLessThanTwoYO() throws Exception {
+		Calendar calendar = Calendar.getInstance();
+		calendar.set(2012, Calendar.FEBRUARY, 2);
```

Comment:
But it also needs to have the visit startDate set to before the estimated birthdata for it to be more meaningful since we are rejecting only dates that come earlier